src/helpers/utils.ts

```diff
@@ -216,7 +216,7 @@ export const getProxyAgent = (proxyOpts?: ProxyConfiguration): ProxyAgent => {
   }
 
   return new ProxyAgent({
-    getProxyForUrl: (url) => {
+    getProxyForUrl: (url: string) => {
```

Comment:
Well, generally it's better to let TypeScript infer arguments this, to not make it lie in the future (e.g. with a dependency bump).

But turns out it's already lying because the `url` **can be `undefined`** (that's why I had put `url?.match()` below).

So I would rather do:

```suggestion
    getProxyForUrl: (url?: string) => {
```

or if not possible:

```suggestion
    getProxyForUrl: (url: string | undefined) => {
```

or if not possible, keep the argument's type implicit.